src/app/client/src/app/modules/shared/components/on-demand-reports/on-demand-reports.component.ts

```diff
@@ -112,17 +107,35 @@ export class OnDemandReportsComponent implements OnInit, OnChanges {
       });
     } else {
       this.isProcessed = true;
-      this.toasterService.error('The request is already in Processed State');
+      this.toasterService.error(_.get(this.resourceService, 'frmelmnts.lbl.requestFailed'));
     }
   }
 
   checkStatus() {
-    const processPendingList = this.onDemandReportData.find(x => x.job_id === this.selectedReport.dataset) || null;
-    if (processPendingList) {
-      return processPendingList['status'] === this.reportStatus.submitted;
-    } else {
-      return false;
+    const selectedReportList = [];
+    _.forEach(this.onDemandReportData, (value) => {
+      if (value.dataset === this.selectedReport.dataset) {
+        selectedReportList.push(value);
+      }
+    });
+    const sortedReportList = _.sortBy(selectedReportList, [(data) => {
+      return data && data.jobStats && data.jobStats.dtJobSubmitted;
+    }]);
+    const reportListData = _.last(sortedReportList) || {};
+    let batchEndDate;
+    if (this.batch.endDate) {
+      batchEndDate = new Date(this.batch.endDate).getTime();
+    }
+    if (!_.isEmpty(reportListData)) {
+      // report is already submitted so dont allow to req again
+      if (reportListData['status'] === this.reportStatus.submitted) {
+        return false;
+      }
+      if (batchEndDate && _.get(reportListData, 'jobStats.dtJobSubmitted ') < batchEndDate) {
```

Comment:
Write different methods to check 
1. Job is already submitted & is in progress. 
ex: isInProgress(reportsdata ) { ... } 
2. Job completed date is less than batch endDate, then only user can request for new one
ex: canRequestReport(reportsData, batchEndDate) { ... }

Write these methods in service. Not in the component.